warehouse/query-core/src/main/java/datawave/query/jexl/functions/ContentFunctionsDescriptor.java

```diff
@@ -300,36 +300,31 @@ public Set<String>[] fieldsAndTerms(Set<String> termFrequencyFields, Set<String>
                         }
                     }
                 } else if (CONTENT_SCORED_PHRASE_FUNCTION_NAME.equals(funcName)) {
-                    JexlNode firstArg = args.next();
-
-                    if (firstArg instanceof ASTNumberLiteral || firstArg instanceof ASTUnaryMinusNode) {
-                        // firstArg is max score value, skip
-                        firstArg = args.next();
-                    }
-
-                    // we override the zones if the first argument is a string
-                    if (firstArg instanceof ASTStringLiteral) {
-                        fields = Collections.singleton(((ASTStringLiteral) firstArg).getLiteral());
-
-                        if (args.peek() instanceof ASTNumberLiteral || args.peek() instanceof ASTUnaryMinusNode) {
-                            args.next(); // max score not needed for fields and terms
-                        }
+                    JexlNode arg = args.next();
 
+                    if (arg instanceof ASTNumberLiteral || arg instanceof ASTUnaryMinusNode) {
+                        // if the first argument is a number, then no field exists
                         termOffsetMap = args.next();
                     } else {
-                        JexlNode nextArg = args.peek();
-
-                        // The zones may (more likely) be specified as an identifier
-                        if (!JexlASTHelper.getIdentifiers(firstArg).isEmpty() && !JexlASTHelper.getIdentifiers(nextArg).isEmpty()) {
-                            if (oredFields != null && firstArg instanceof ASTAndNode) {
-                                oredFields.setValue(false);
-                            }
-
-                            fields = JexlASTHelper.getIdentifierNames(firstArg);
-                            termOffsetMap = args.next();
+                        if (arg instanceof ASTIdentifier) {
+                            // single field case
+                            fields = Collections.singleton(String.valueOf(JexlASTHelper.getIdentifier(arg)));
                         } else {
-                            termOffsetMap = firstArg;
+                            // multi field case
```

Comment:
it might be helpful to add an "e.g.," here and for the single field case above, such as:

```
// multi field case, e.g., content:scoredPhrase((FIELD_A || FIELD_B), -1.5, termOffsetMap, 'foo', 'bar')
```

Not critical b/c there are examples in the unit test, but it makes the code a bit more readable.

Reply:
I can absolutely do that